snuba/utils/streams/synchronized.py

```diff
@@ -86,8 +90,18 @@ def __init__(
         ] = Synchronized({group: {} for group in commit_log_groups})
 
         self.__commit_log_worker_stop_requested = Event()
+        self.__commit_log_worker_subscription_received = Event()
         self.__commit_log_worker = execute(self.__run_commit_log_worker)
 
+        logger.debug("Waiting for commit log consumer to receieve assignment...")
+        while not self.__commit_log_worker_subscription_received.wait(0.1):
+            # Check to make sure we're not waiting for an event that will never
+            # happen if the commit log consumer has crashed.
+            if not self.__commit_log_worker.running():
+                self.__commit_log_worker.result()
+        else:
+            logger.debug("Commit log consumer has started.")
```

Comment:
This will slow down subscriptions startup because it waits for two assignments serially rather than in parallel. An alternative would be exposing the ability to wait for this event publically for the sake of tests, but that seems a bit like breaking encapsulation?

Reply:
We're not talking about a significant delay here, though, right?

Reply:
I don't have any empirical evidence from production, but I'd expect this to take single to double digit seconds.

Reply:
Yeah.. I'm not super sad about that.